Modules/faulthandler.c

```diff
@@ -212,6 +216,80 @@ faulthandler_dump_traceback(int fd, int all_threads,
     reentrant = 0;
 }
 
+#ifdef HAVE_EXECINFO_H
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+#define BACKTRACE_SIZE 16
+#define TRACEBACK_ENTRY_MAX_SIZE 256
+    void *callstack[BACKTRACE_SIZE];
+    int frames = backtrace(callstack, BACKTRACE_SIZE);
+    char **strings = backtrace_symbols(callstack, BACKTRACE_SIZE);
+    if (strings == NULL) {
+        PUTS(fd, "  <failed to get stack trace>\n");
+    }
+    else {
+        for (int i = 0; i < frames; ++i) {
+            char entry_str[TRACEBACK_ENTRY_MAX_SIZE];
+            snprintf(entry_str, TRACEBACK_ENTRY_MAX_SIZE, "  %s\n", strings[i]);
+            size_t length = strlen(entry_str) + 1;
+            if (length == TRACEBACK_ENTRY_MAX_SIZE) {
+                /* We exceeded the size, make it look prettier */
+
+                // Add ellipsis to last 3 characters
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 5] = '.';
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 4] = '.';
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 3] = '.';
+
```

Comment:
Maybe you can reduce the number of blank lines. I think the comments sufficiently separate the code but it's also up to you.